ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -65,45 +65,49 @@ -(void)uploadResult{
 - (void)doUploadWithProgress {
     if (self.result == nil && self.measurement == nil && self.upload_all) {
         //upload ALL
-        SRKResultSet *notUploaded = [Measurement notUploadedMeasurements];
-        [self uploadMeasurements:notUploaded];
+        [self uploadMeasurements:[Measurement notUploadedMeasurements]];
     }
     else if (self.result != nil && self.measurement == nil && self.upload_all) {
         //upload all measurements of that result
-        SRKResultSet *notUploaded = [self.result notUploadedMeasurements];
-        [self uploadMeasurements:notUploaded];
+        [self uploadMeasurements:[self.result notUploadedMeasurements]];
     }
     else if (self.result != nil && self.measurement != nil && !self.upload_all) {
         //upload this measurement
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [MBProgressHUD HUDForView:self.navigationController.view].label.text = NSLocalizedFormatString(@"Modal.ResultsNotUploaded.Uploading", @"");
-        });
-        [self uploadSingleMeasurement:self.measurement];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [MBProgressHUD HUDForView:self.navigationController.view].progress = 1;
-        });
+        [self uploadMeasurements:[NSArray arrayWithObject:self.measurement]];
     }
 }
 
--(void)uploadMeasurements:(SRKResultSet *)notUploaded{
+-(void)uploadMeasurements:(NSArray *)notUploaded {
+    [self uploadMeasurements:notUploaded startAt:0];
+}
+
+//SRKResultSet is a subclass of NSArray
+-(void)uploadMeasurements:(NSArray *)notUploaded startAt:(int)idx{
     if ([notUploaded count] == 0) return;
     float progress = 0.0f;
     float measurementValue = 1.0/[notUploaded count];
-    int done = 1;
-    for (Measurement *currentMeasurement in notUploaded){
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [MBProgressHUD HUDForView:self.navigationController.view].label.text = NSLocalizedFormatString(@"Modal.ResultsNotUploaded.Uploading", [NSString stringWithFormat:@"%d/%ld", done, [notUploaded count]]);
-        });
-        [self uploadSingleMeasurement:currentMeasurement];
-        progress += measurementValue;
-        done++;
+    while (idx < [notUploaded count]){
```

Comment:
I think the type of `idx` should be the same type returnd by `[notUploaded count]`

Reply:
it is the same, they are both int

Reply:
Are you sure? NSUInteger strikes me as an _unsigned_ integer:

https://developer.apple.com/documentation/foundation/nsarray/1409982-count?language=objc

Reply:
I don't really know good these obj-c types but surely I will rename it